src/main/java/com/yelp/nrtsearch/server/grpc/LuceneServer.java

```diff
@@ -1545,7 +1545,9 @@ public void recvRawFile(
                     .build();
             rawFileChunkStreamObserver.onNext(rawFileChunk);
             totalRead += chunkSize;
-            randomDelay(random);
+            if (globalState.configuration.getFileSendDelay()) {
```

Comment:
we should probably do a backfill of YelpReviewsDataset (or some other dataset) with the delay turned off. IIRC I saw this issue when backfilling an entire index with high throughput and primary buffer OOMed since replica could not keep up with primary producing data.